### PR TITLE
Update cluster-api-release-team github team

### PR DESCRIPTION
**Pull request by joekr**
This PR updates the team called the cluster-api-release-team.

### CODE REVIEW

**Comment by fabriziopandini** [comment]
> The following members of team cluster-api-release-team are not Kubernetes SIGs org members: adilghaffardev, sadysnaat

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml [hunk 1]
```diff
@@ -270,20 +270,20 @@ teams:
...
     - furkatgofurov7
+    - g-gaston
     - hackeramitkumar
     - joekr
+    - johannesfrey
+    - killianmuldoon
     - nawazkh
-    - oscr
-    - sayantani11
+    - prajyot-parab
```
Comment (CecileRobertMichon) [inline config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:284]: not sure if it matters but @Prajyot-Parab's username is upper cased

Comment (sbueringer) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:284]: Looks like for VibhorChinda we used the actual casing, so we should change it (does this apply to anyone else as well?

Comment (CecileRobertMichon) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:284]: I checked everyone else, this is the only one that was inconsistent

Comment (joekr) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml]: good call. I've check all the other names for casing changes.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**